playbooks/compliance_check.yaml

```diff
@@ -4,35 +4,33 @@
   tasks:
     - name: Run compliance script
       ansible.builtin.shell: |
-        python3 ~/Tests/scs-compliance-check.py ~/Tests/scs-compatible-iaas.yaml -s {{ cloud }} -a os_cloud={{ cloud }} -o report.yaml -C
-      register: result
+        python3 ~/Tests/cleanup.py -c {{ item }} --prefix _scs-
+        python3 ~/Tests/scs-compliance-check.py ~/Tests/scs-compatible-iaas.yaml -s {{ item }} -a os_cloud={{ item }} -o report-{{ item }}.yaml -C
+        ssh-keygen -Y sign -f ~/id_subject -n report report-{{ item }}.yaml
+        curl --data-binary @report-{{ item }}.yaml.sig --data-binary @report-{{ item }}.yaml -H "Content-Type: application/x-signed-yaml" -H "Authorization: Basic $AUTH_TOKEN" https://compliance.sovereignit.cloud/reports
       changed_when: true
       failed_when: false
       # ^^^ this task does not fail; the only failure would be if `report.yaml` didn't get produced,
```

Comment:
the comment is correct, but is it a desired behavior? Failure (i.e. outage) of single cloud could lead to absence of the report for all. Previously different targets were checked by individual jobs. With this change it is not the case anymore and that would lead to the problem. There are few ways how to deal with it in an "Ansible" way:

- stat the expected reports and only sync/return existing ones (something like https://opendev.org/openstack/codegenerator/src/branch/master/playbooks/rust/post.yaml#L46-L72)
- put the "compliance" as such into the ansible role and invoke it in the loop. That helps you to focus on single entity in the tasks instead of messing with loop in every task

Generally I think a combination of both should be done since loops in a main playbook tend to turn into disaster pretty quickly